### ISSUE TITLE
test coverage

### ISSUE THREAD

**Issue by nschloe** [krober10nd/SeismicMesh#36]
With under 60%, SeismicMesh has a very low code coverage. In my experience, code that isn't tested doesn't work (or at least doesn't work much of the time).

![13e](https://user-images.githubusercontent.com/181628/91722401-7b0a8780-eb9a-11ea-9109-f25fa9ffea3b.jpg)

You could cover much ground for example by testing (or removing of course) your `plot` functions (see https://codecov.io/gh/krober10nd/SeismicMesh/src/par3d/SeismicMesh/sizing/mesh_size_function.py). Since your tests run headless, you best set the env variable `MPLBACKEND=Agg`. Locally, I always run my tests with
```
MPLBACKEND=Agg pytest --maxfail=1
```
Perhaps that's useful for you as well.

**Comment by krober10nd** [comment]
Hahah :] That's awesome. 

Yes, plotting is a bit under tested. But the real culprit has always been the MPI parallelism. I wasn't able to run the parallel tests on the CI server, which knocks down the coverage by around 20%.

**Comment by krober10nd** [comment]
when I do ``mpirun -np 2 pytest tests/ -m "parallel"``  it literally runs two pytests

**Comment by nschloe** [comment]
>  But the real culprit has always been the MPI parallelism.

Ah, I see how this can be a problem. I'd definitely love to see you look into this one more time.

**Comment by krober10nd** [comment]
Ok investigating.
when I call the tests like, it works fine locally capturing both serial and parallel tests.

``
     mpirun -np 2 python -m pytest --cov {envsitepackagesdir}/SeismicMesh --cov-report xml --cov-report term
``

However on the circleCI serve, I get: 

``18269,1],0]: A high-performance Open MPI point-to-point messaging module
was unable to find any relevant network interfaces:``

https://app.circleci.com/pipelines/github/krober10nd/SeismicMesh/516/workflows/4d48eb16-31ff-4225-b92c-70f3f519630d/jobs/1019

**Comment by nschloe** [comment]
Ah yes, I've seen this before. (gmsh unnecessarily had mpi compiled in, I resolved the issue by fixing gmsh to _not_ link against mpi anymore.) Perhaps circleci isn't suited for MPI. I'd search/ask in [their forums](https://discuss.circleci.com/). You might also want to try out another ci provider; travis and github-actions come to mind.

**Comment by krober10nd** [comment]
Okay, after a little investigation I figured it out how to run my parallel tests using CircleCI in #39 

* My issue was that I was running some tests that were only meant for serial execution in parallel so I stopped doing that and now it works. There was also a way to suppress the warning I wrote in the above post.

* I deprecated the `plots` module; they weren't used anyway I had just made some figures once with them but I've since found better plotting alternatives. 

* Added a new 3d parallel test that runs what I call the `blocker` decomp module. This takes an initial point distribution, localizes it in parallel, and does some meshing iterations. I test decomposing all three axes. Down the road, I plan to use this more for mesh adaptation. 

*  According to me, test coverage is up by ~16% and hovering around 76%. Having a look at the coverage report shows that the `geometry` modules needs to be tested more.
